Feature 673 release_guide_met

JohnHalleyGotway opened the pull request:
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Adding first draft of release steps for MET along with consistency updates.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Read through 3 versions of release steps for MET. Consider whether "release_steps/met/create_release_on_github.rst" is general enough to be applied to the other METplus repos.

- [x] Do these changes include sufficient documentation and testing updates? **[Yes]**
That's what they are!

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, and **Milestone**
- [x] After submitting the PR, select **Linked Issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

JohnHalleyGotway committed:
Per #673, rename close_release_issue to finalize_release_on_github with instructions broken out for development, official, and bugfix release.

JohnHalleyGotway committed:
Per #673, work in progress for MET instructions.

JohnHalleyGotway committed:
Per #673 more work.

JohnHalleyGotway committed:
Per #673, more updates to MET instructions.

JohnHalleyGotway committed:
Per #673, more updates.

jprestop commented on docs/Release_Guide/release_steps/met/create_release_on_github.rst:
```diff
@@ -1,9 +1,14 @@
+
+* Set the branch value to 'develop' for a development release or 'main_vX.Y' for a bugfix or official release
+
+* Choose a release name (i.e. |projectRepo|-X.Y.Z, |projectRepo|-X.Y.Z-betaN, or |projectRepo|-X.Y.Z)
```
```suggestion
* Choose a release name (i.e. |projectRepo|-X.Y.Z-betaN, |projectRepo|-X.Y.Z-rcN, or |projectRepo|-X.Y.Z)
```
I noticed that the first and last items in "i.e" were the same and was wondering if it should be as I suggest here (betaN, rcN, empty)?

jprestop committed:
Update update_docs.rst

Changed "-" to "*" to be consistent with other documentation and more Sphinx-like.

jprestop commented on docs/Release_Guide/release_steps/met/update_version_development.rst:
```diff
@@ -1,22 +1,18 @@
-*  Work on a feature branch from the *develop* branch being sure to include
-   the GitHub issue number for the new release
-   (e.g. feature_NNNN_vX.Y.Z-betaN or feature_NNNN_vX.Y-rcN).
+*  Create a feature branch from the *develop* branch being sure to include the GitHub issue number for the new release (e.g. feature_NNNN_vX.Y.Z-betaN or feature_NNNN_vX.Y-rcN).
```
```suggestion
*  Create a feature branch from the *develop* branch being sure to include the GitHub issue number for the new release (e.g. feature_NNNN_vX.Y.Z-betaN or feature_NNNN_vX.Y.Z-rcN).
```
Should the rcN release have a Z also?

jprestop commented on docs/Release_Guide/release_steps/met/update_version_official.rst:
```diff
@@ -1,13 +1,28 @@
+
+    * Test configuration files, searching recursively, in *test/config*.
+
+  * If necessary, add a new *met/data/table_files/met_header_columns_VX.Y.txt* defining the columns names for this version.
```
Should met_header_columns_VX.Y.txt have a ".Z" or does it stay "X.Y"?

JohnHalleyGotway replied:
Yes, X.Y is correct. The headers do not change for bugfix releases... so we only need to keep track of X.Y, not X.Y.Z.

jprestop commented on docs/Release_Guide/release_steps/met/update_version_official.rst:
```diff
@@ -1,13 +1,28 @@
+
+  * If necessary, add a new *met/data/table_files/met_header_columns_VX.Y.txt* defining the columns names for this version.
+
+  * If necessary, add a new *test/hdr/met_X_Y.hdr* file defining the column names for this version for the test scripts.
```
Should met_X_Y.hdr*  have a".Z" or does it stay "X.Y"?

JohnHalleyGotway replied:
Yes, X.Y is correct. The headers do not change for bugfix releases... so we only need to keep track of X.Y, not X.Y.Z.

JohnHalleyGotway committed:
Update docs/Release_Guide/release_steps/met/create_release_on_github.rst

Co-authored-by: jprestop <jpresto@ucar.edu>

JohnHalleyGotway committed:
Update docs/Release_Guide/release_steps/met/update_version_development.rst

Co-authored-by: jprestop <jpresto@ucar.edu>

JohnHalleyGotway committed:
Per #673, work on the first page of teh release guide.

jprestop reviewed (approved):
Changes are clear and look great!
